node-maintainer/src/main/java/com/yahoo/vespa/hosted/node/certificate/CertificateAuthorityClient.java

```diff
@@ -0,0 +1,88 @@
+package com.yahoo.vespa.hosted.node.certificate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yahoo.vespa.hosted.athenz.instanceproviderservice.ca.model.CertificateSerializedPayload;
+import com.yahoo.vespa.hosted.athenz.instanceproviderservice.ca.model.CsrSerializedPayload;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLContextBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Sends Certificate Signing Requests to config server
+ *
+ * @author freva
+ */
+class CertificateAuthorityClient {
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final String caServerPath;
```

Comment:
Consider changing type to {{URI}}. The name is somewhat confusing as it might be interpreted at the path part of an uri.